cypress/integration/vesselCommunications.spec.ts

```diff
@@ -1,57 +1,231 @@
-
-describe("As a beacon owner, I want to register my communication details so SAR can contact me in an emergency", () => {
+import {
+  andIClickContinue,
+  finallyIClear,
+  finallyIUncheck,
+  givenIAmAt,
+  givenIHaveSelected,
+  iCanClickTheBackLinkToGoToPreviousPage,
+  thenIShouldSeeAnErrorMessageThatContains,
+  thenIShouldSeeAnErrorSummaryLinkThatContains,
+  thenMyFocusMovesTo,
+  thenTheUrlShouldContain,
+  whenIClickContinue,
+  whenIClickOnTheErrorSummaryLinkContainingText,
+  whenIType,
+} from "./common.spec";
+
+describe("As a beacon owner and maritime pleasure vessel user", () => {
   const pageUrl = "/register-a-beacon/vessel-communications";
 
   beforeEach(() => {
-    givenIAmOnTheVesselCommunicationsPage();
+    givenIAmAt(pageUrl);
   });
 
-  it("requires an MMSI number if the fixed VHF checkbox is selected", () => {
-    givenIHaveSelected("#fixedVhfRadio");
-    andIHaveLeftTheRelevantTextInputBlank();
-
-    whenIClickContinue();
+  describe("the Fixed VHF/DSC radio option", () => {
+    it("requires an MMSI number if the fixed VHF/DSC radio checkbox is selected", () => {
+      const expectedErrorMessage = ["We need", "fixed VHF"];
+      givenIHaveSelected("#fixedVhfRadio");
+
+      whenIType(" ", "fixedVhfRadioInput");
+      whenIClickContinue();
+      thenIShouldSeeAnErrorSummaryLinkThatContains(...expectedErrorMessage);
+      thenIShouldSeeAnErrorMessageThatContains(...expectedErrorMessage);
+      whenIClickOnTheErrorSummaryLinkContainingText(...expectedErrorMessage);
+      thenMyFocusMovesTo("fixedVhfRadioInput");
+
+      finallyIClear("#fixedVhfRadioInput");
+      finallyIUncheck("#fixedVhfRadio");
+    });
+
+    it("requires the fixed MMSI number to be 9 characters long", () => {
+      const expectedErrorMessage = ["MMSI number", "nine digits long"];
+      givenIHaveSelected("#fixedVhfRadio");
+
+      whenIType("012345678910", "fixedVhfRadioInput");
+      andIClickContinue();
+
+      thenIShouldSeeAnErrorSummaryLinkThatContains(...expectedErrorMessage);
+      thenIShouldSeeAnErrorMessageThatContains(...expectedErrorMessage);
+      whenIClickOnTheErrorSummaryLinkContainingText(...expectedErrorMessage);
+      thenMyFocusMovesTo("fixedVhfRadioInput");
+
+      finallyIClear("#fixedVhfRadioInput");
+      finallyIUncheck("#fixedVhfRadio");
+    });
+
+    it("requires the fixed MMSI number to be numbers 0 to 9 only", () => {
+      const expectedErrorMessage = ["MMSI number", "numbers", "0 to 9"];
+      givenIHaveSelected("#fixedVhfRadio");
+
+      whenIType("9charslng", "fixedVhfRadioInput");
+      andIClickContinue();
+
+      thenIShouldSeeAnErrorSummaryLinkThatContains(...expectedErrorMessage);
+      thenIShouldSeeAnErrorMessageThatContains(...expectedErrorMessage);
+      whenIClickOnTheErrorSummaryLinkContainingText(...expectedErrorMessage);
+      thenMyFocusMovesTo("fixedVhfRadioInput");
+
+      finallyIClear("#fixedVhfRadioInput");
+      finallyIUncheck("#fixedVhfRadio");
```

Comment:
To confirm, are we clearing the input here as the browser is not refreshing the page and therefore the input isn't cleared, so each test is responsible for doing that?

Reply:
I've removed these from the test suite in favour of reloading the page in the `beforeEach` hook.  The performance improvement actually came from not having to go to `/` to set the cookie because the cookie is stubbed now, so this hasn't actually slowed down the test suite.  🙈

But it did cause flakiness, so now a page reload happens to reset state.